src/uv-common.c

```diff
@@ -155,6 +155,22 @@ static const char* uv__unknown_err_code(int err) {
   return copy != NULL ? copy : "Unknown system error";
 }
 
+#define UV_ERR_NAME_GEN_R(name, _) case UV_ ## name: { strncpy(buf, #name, buflen); break; }
+const char* uv_err_name_r(int err, char *buf, size_t buflen) {
+  switch (err) {
+    UV_ERRNO_MAP(UV_ERR_NAME_GEN_R)
+    default: {
+      snprintf(buf, buflen, "Unknown system error %d", err);
+      break;
+    }
+  }
+  /* ensure zero-termination */
+  if ((buf != NULL) && (buflen != 0))
+    buf[buflen-1] = '\0';
```

Comment:
Not necessary, `snprintf()` always zero-terminates, unless `len == 0` (which should probably be an error.)